pkg/providers/httpjson/httpjson.go

```diff
@@ -0,0 +1,138 @@
+package httpjson
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/antchfx/jsonquery"
+
+	"github.com/helmfile/vals/pkg/api"
+	"github.com/helmfile/vals/pkg/log"
+)
+
+type provider struct {
+	// Keeping track of httpjson services since we need a service per url
+	protocol   string
+	log        *log.Logger
+	docs       map[string]*jsonquery.Node
+	floatAsInt bool
+}
+
+func New(l *log.Logger, cfg api.StaticConfig) *provider {
+	p := &provider{
+		log: l,
+	}
+
+	// Should the protocol be insecure i.e. http
+	insecureArg := cfg.String("insecure")
+	p.protocol = "https"
+	if insecureArg == "true" {
+		p.protocol = "http"
+	}
+
+	// By default JSON will return large integers as float64
+	floatAsIntArg := cfg.String("floatAsInt")
+	p.floatAsInt = false
+	if floatAsIntArg == "true" {
+		p.floatAsInt = true
+	}
+
+	// Initialize docs map to store the json object for use multiple times
+	if len(p.docs) == 0 {
+		p.docs = make(map[string]*jsonquery.Node)
+	}
+
+	return p
+}
+
+func GetXpathFromUri(uri string) (xpath string, err error) {
```

Comment:
what will happen if uri is empty? and add unittest for function as possible as you can. thanks so much.

Reply:
Added unit testing 🙏